main.md

```diff
@@ -193,180 +199,327 @@ Note: Applications MUST ensure that their authorization data types do not collid
 The following example shows how an implementation could utilize the namespace `https://scheme.example.org/` to ensure collision resistant element names.
 
 ```JSON
-{ 
-   "type":"https://scheme.example.org/files",
-   "locations":[ 
+{
+   "type": "https://scheme.example.org/files",
+   "locations": [
       "https://example.com/files"
    ],
-   "permissions":[ 
-      { 
-         "path":"/myfiles/A",
-         "access":[ 
+   "permissions": [
+      {
+         "path": "/myfiles/A",
+         "access": [
             "read"
          ]
       },
-      { 
-         "path":"/myfiles/A/X",
-         "access":[ 
+      {
+         "path": "/myfiles/A/X",
+         "access": [
             "read",
             "write"
          ]
       }
    ]
 }
 ```
+## Relationship to "scope" parameter {#scope}
+
+`authorization_details` and `scope` can be used in the same authorization request for carrying independent authorization requirements. 
+
+The AS MUST keep both requirements sets separate when reflecting this data to the client and passing this data through to resource servers.
+
+For user convenience, the AS is supposed to merge the requirements when asking the user for consent.
```

Comment:
This isn't just for convenience, this is for security: if I ask for a `scope` that's got more access than the `details` object asks for, it should be that the result is both of them together. Otherwise we can end up with naive implementations giving more than they intend, I think.

Reply:
Even if the AS does not merge the authz requirements it must ask for consent for every detail. I don't see how this could result in security issues.

Besides this, the text states that scopes and authz details shall be independent. So you can use openid as scope but the authz details shall not contain openid stuff. I know that's pretty restrictive but I also think it's acceptable and (I hope) makes things easier.

Reply:
I don't think that's  a testable restriction, to be honest. And we're going to end up with some API that wants to break that. 

The security issue comes from potentially opening things wider than one intended.

Reply:
So your proposal is we should accept that implementers use both channels to pass requirements re the same API.  I'm fine with it as long as we can define what this means. Can you please propose text?

Reply:
```suggestion
When gathering user consent, the AS MUST present the merged set of requirements represented by the authorization request. 
```

Reply:
How about "complete" rather than "merged"? Trying to say that you gotta show em all somehow but not necessarily saying how. The word "merged" to me kinda suggests some processing.

Reply:
```suggestion
When gathering user consent, the AS MUST present the complete set of requirements represented by the authorization request. 
```

Reply:
I also like "complete" better:

```suggestion
When gathering user consent, the AS MUST present the complete set of requirements represented by the authorization request. 
```